### PR TITLE
integer literal documentation [ci skip]

### PR DESCRIPTION
**Pull request by krux02**
fixes https://github.com/nim-lang/Nim/issues/12503

### CODE REVIEW

File: doc/intern.rst [hunk 1]
```diff
@@ -667,3 +667,91 @@ important the hidden formal param is ``void*`` and not something more
...
+pointing back to the integer literal node in the ast containing the
+integer value. These are the properties that hold true for integer
+literal types.
+
+    n.kind == nkIntLit
+    n.typ.kind == tyInt
+    n.typ.n == n
+
+Other literal types, such as `uint literal(123)` that would
+automatically convert to other integer types, but refuses prefers to
```
Comment (Jjp137) [inline doc/intern.rst]: "but refuses prefers to" -- did you mean to remove "prefers" or...?

File: doc/intern.rst [hunk 2]
```diff
@@ -667,3 +667,91 @@ important the hidden formal param is ``void*`` and not something more
 specialized. However the more specialized env type needs to passed to the
 backend somehow. We deal with this by modifying ``s.ast[paramPos]`` to contain
 the formal hidden parameter, but not ``s.typ``!
+
+
+Integer literals:
+-----------------
+
+In Nim, there is a redundant way to specify declare the type of an
```
Comment (narimiran) [inline doc/intern.rst]: > to _specify declare_ the type

One of these words is unneeded?

File: doc/intern.rst (hunk 1 above)
Comment (krux02) [reply doc/intern.rst]: I meant prefers. Change has been made. Thank you.

File: doc/intern.rst (hunk 2 above)
Comment (krux02) [reply doc/intern.rst]: yes it is. Change is done.

**Pull request merged**